### PR TITLE
BE&FE

### PR DESCRIPTION
**Pull request by alyoona**
MainPage view mode
LoginPage
Participant-Go
ParticipantsPage(preview for admin)

### CODE REVIEW

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java [hunk 1]
```diff
@@ -7,22 +7,35 @@
...
         this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
     }
 
     @Override
-    public Conference getById(Long id) {
+    public Optional<Conference> getById(Long id) {
         MapSqlParameterSource mapSqlParameter =new MapSqlParameterSource();
         mapSqlParameter.addValue("id", id);
-        return namedParameterJdbcTemplate.queryForObject(GET_CONFERENCE_SQL, mapSqlParameter, conferenceMapper);
+        return namedParameterJdbcTemplate.query(GET_CONFERENCE_SQL, mapSqlParameter, conferenceMapper).stream().findFirst();
```
Comment (groksana) [inline src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: why findFirst if id is unique?

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java [hunk 2]
```diff
@@ -7,22 +7,35 @@
...
+    }
+
+    @Override
+    public Optional<Conference> getNearestConference() {
+        return namedParameterJdbcTemplate.query(GET_CONFERENCE_BY_MAX_ID_SQL, conferenceMapper).stream().findFirst();
```
Comment (groksana) [inline src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: findFirst again?

File: src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java [hunk 3]
```diff
@@ -16,6 +16,15 @@ public ConferenceServiceImpl(ConferenceRepository conferenceRepository) {
...
     @Override
     public Conference getById(Long id) {
-        return conferenceRepository.getById(id);
+
+        return conferenceRepository.getById(id).orElse(null);
+    }
+
+    @Override
+    public Conference getNearestConference() {
+        return conferenceRepository.getNearestConference().orElse(null);
```
Comment (groksana) [inline src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java]: I expect some exception

File: src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java [hunk 4]
```diff
@@ -16,6 +16,15 @@ public ConferenceServiceImpl(ConferenceRepository conferenceRepository) {
 
     @Override
     public Conference getById(Long id) {
-        return conferenceRepository.getById(id);
+
+        return conferenceRepository.getById(id).orElse(null);
```
Comment (groksana) [inline src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java]: I expect some exception like:
Conference with id 5 is not found

File: src/main/java/com/jland/website/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.jland.website.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private UserRepository userRepository;
+    @Autowired
```
Comment (groksana) [inline src/main/java/com/jland/website/service/impl/UserServiceImpl.java]: you can remove autowired

File: src/main/java/com/jland/website/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    public UserServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User findUserByUsername(String username) {
+        return userRepository.findByUsername(username);
+    }
```
Comment (groksana) [inline src/main/java/com/jland/website/service/impl/UserServiceImpl.java]: any exception when user is not found?

File: src/main/java/com/jland/website/web/controller/HomePageController.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+public class HomePageController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HomePageController.class);
+
+    private final ConferenceService conferenceService;
+    private final EventService eventService;
+    private final ParticipantService participantService;
+    private final UserService userService;
+
+    @Autowired
```
Comment (groksana) [inline src/main/java/com/jland/website/web/controller/HomePageController.java]: you can remove autowired

File: src/main/java/com/jland/website/web/controller/ParticipantController.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
+package com.jland.website.web.controller;
+
+import com.jland.website.service.ParticipantService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping(path = "/conference/{conferenceId}")
```
Comment (groksana) [inline src/main/java/com/jland/website/web/controller/ParticipantController.java]: should be /conferences/{conferenceId}

File: src/main/java/com/jland/website/web/controller/ParticipantController.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+public class ParticipantController {
+
+    private ParticipantService participantService;
+
+    @Autowired
```
Comment (groksana) [inline src/main/java/com/jland/website/web/controller/ParticipantController.java]: please remove

File: src/main/java/com/jland/website/web/controller/ParticipantController.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping(path = "/conference/{conferenceId}")
+public class ParticipantController {
+
+    private ParticipantService participantService;
```
Comment (groksana) [inline src/main/java/com/jland/website/web/controller/ParticipantController.java]: better final

File: src/main/resources/property/db.properties [hunk 11]
```diff
@@ -1,5 +1,13 @@
...
+
+#Heroku Database Credentials
 postgres.db.driverclassname=org.postgresql.Driver
-postgres.db.url=jdbc:postgresql://localhost:5432/jland
-postgres.db.username=admin
-postgres.db.password=admin
+postgres.db.url=jdbc:postgresql://ec2-46-137-121-216.eu-west-1.compute.amazonaws.com:5432/d1vrp9hrsk2i9v?sslmode=require
+postgres.db.username=qcjppuyykaxfdt
+postgres.db.password=1944d4ef94c94f3cf89f03888818af495fa9eeefd19257802ba5b25e09b514c2
 postgres.db.poolsize=3
```
Comment (groksana) [inline src/main/resources/property/db.properties]: I think it's to git ignore

File: src/test/java/com/jland/website/repository/mapper/PresentationMapperTest.java [hunk 12]
```diff
@@ -20,23 +20,23 @@
     void mapRow() throws SQLException {
         when(resultSet.getLong("presentation_id")).thenReturn(1L);
         when(resultSet.getString("presentation_name")).thenReturn("presentation");
-        when(resultSet.getLong("event_id")).thenReturn(2L);
+/*        when(resultSet.getLong("event_id")).thenReturn(2L);
         when(resultSet.getString("event_name")).thenReturn("event");
         when(resultSet.getString("type")).thenReturn("break");
         when(resultSet.getTime("start_time")).thenReturn(Time.valueOf(LocalTime.of(12, 0)));
-        when(resultSet.getTime("end_time")).thenReturn(Time.valueOf(LocalTime.of(13, 0)));
+        when(resultSet.getTime("end_time")).thenReturn(Time.valueOf(LocalTime.of(13, 0)));*/
```
Comment (groksana) [inline src/test/java/com/jland/website/repository/mapper/PresentationMapperTest.java]: commented code??

Comment (alyoona) [reply src/test/java/com/jland/website/repository/mapper/PresentationMapperTest.java]: removed

File: src/main/java/com/jland/website/web/controller/ParticipantController.java (hunk 8 above)
Comment (alyoona) [reply src/main/java/com/jland/website/web/controller/ParticipantController.java]: done

File: src/main/java/com/jland/website/web/controller/ParticipantController.java (hunk 10 above)
Comment (alyoona) [reply src/main/java/com/jland/website/web/controller/ParticipantController.java]: done

File: src/main/java/com/jland/website/config/security/WebSecurityConfig.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        auth.userDetailsService(userDetailsServiceImpl)
+                .passwordEncoder(encoder())
+        ;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http.authorizeRequests()
+                .antMatchers("/assets/**").permitAll()
```
[the remark this answers is not in the document]
Comment (alyoona) [reply src/main/java/com/jland/website/config/security/WebSecurityConfig.java]: as for "/assets" has discussed, it stays as is

File: src/main/java/com/jland/website/config/security/WebSecurityConfig.java [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+                .formLogin()
+                    .loginPage("/login").permitAll()
+                    .defaultSuccessUrl("/home", true)
+                .and()
+                .logout().permitAll()
+               ;
+    }
+
+    @Autowired
+    public void setUserDetailsServiceImpl(UserDetailsServiceImpl userDetailsServiceImpl) {
```
[the remark this answers is not in the document]
Comment (alyoona) [reply src/main/java/com/jland/website/config/security/WebSecurityConfig.java]: changed to constructor

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java (hunk 1 above)
Comment (alyoona) [reply src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: changed

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java (hunk 2 above)
Comment (alyoona) [reply src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: changed

File: src/main/java/com/jland/website/web/controller/ParticipantController.java (hunk 9 above)
Comment (alyoona) [reply src/main/java/com/jland/website/web/controller/ParticipantController.java]: removed

File: src/main/java/com/jland/website/model/Event.java [hunk 15]
```diff
@@ -6,12 +6,12 @@
 public class Event {
 
     private long id;
-    private Conference conference;
     private String name;
     private String type;
     private LocalTime startTime;
     private LocalTime endTime;
     private int order;
```
[the remark this answers is not in the document]
Comment (alyoona) [reply src/main/java/com/jland/website/model/Event.java]: added not null constraint to event.event_order

File: src/main/java/com/jland/website/web/controller/HomePageController.java (hunk 7 above)
Comment (alyoona) [reply src/main/java/com/jland/website/web/controller/HomePageController.java]: removed

File: src/main/java/com/jland/website/service/impl/UserServiceImpl.java (hunk 5 above)
Comment (alyoona) [reply src/main/java/com/jland/website/service/impl/UserServiceImpl.java]: removed

File: src/main/java/com/jland/website/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (alyoona) [reply src/main/java/com/jland/website/service/impl/UserServiceImpl.java]: added

File: src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java (hunk 4 above)
Comment (alyoona) [reply src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java]: added

File: src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java (hunk 3 above)
Comment (alyoona) [reply src/main/java/com/jland/website/service/impl/ConferenceServiceImpl.java]: added

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java (hunk 1 above)
Comment (alyoona) [reply src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: changed

File: src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java (hunk 2 above)
Comment (alyoona) [reply src/main/java/com/jland/website/repository/jdbc/JdbcConferenceRepository.java]: changed

File: src/main/resources/property/db.properties (hunk 11 above)
Comment (alyoona) [reply src/main/resources/property/db.properties]: done

**Pull request merged**